crates/uv/src/commands/tool/run.rs

```diff
@@ -424,7 +424,7 @@ async fn get_or_create_environment(
     };
 
     // Check if the tool is already installed in a compatible environment.
-    if !isolated && settings.reinstall.is_none() && settings.upgrade.is_none() {
+    if !isolated && settings.reinstall.is_none() {
```

Comment:
I mean... should `--reinstall` get the same treatment?

Reply:
Yeah I don't think these flags really make sense here?

Reply:
It does have _some_ effect though. `--reinstall` would cause us to reinstall all packages in the _cached_ environment.

Reply:
I feel like the cached environment shouldn't be mutable to the user — it should be all or nothing with `--refresh` covering a reset of it or `--isolated` bypassing it?

Reply:
Basically, I think we should treat the cached item as immutable. If we need to update it, we should discard it.

Reply:
How should the user indicate that it should be re-created?

Reply:
`--refresh`, I think.